keyboards/feker/ik75/keymaps/default/keymap.c

```diff
@@ -0,0 +1,267 @@
+       ├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────┬───┘ ├───┤
+       │        │   │   │   │   │   │   │   │   │   │   │      │┌───┐│   │
+       ├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬─────┘│   │└───┘
+       │    │    │    │                        │    │    │  ┌───┼───┼───┐
+       └────┴────┴────┴────────────────────────┴────┴────┘  │   │   │   │
+                                                            └───┴───┴───┘
+*/
+    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14                15              */
+    [_FN2] = LAYOUT(
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+                _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,          _______,
+                _______, _______, _______,                            _______,                   _______, _______,          _______, _______, _______,          _______
+            ),
+};
+
+void keyboard_post_init_user(void) {
+    user_config.raw = eeconfig_read_user();
+    switch (user_config.rgb_mode) {
+        case RGB_MODE_ALL:
+            rgb_matrix_set_flags(LED_FLAG_ALL);
+            rgb_matrix_enable_noeeprom();
+            break;
+        case RGB_MODE_KEYLIGHT:
+            rgb_matrix_set_flags(LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER | LED_FLAG_INDICATOR);
+            rgb_matrix_set_color_all(0, 0, 0);
+            break;
+        case RGB_MODE_UNDERGLOW:
+            rgb_matrix_set_flags(LED_FLAG_UNDERGLOW);
+            rgb_matrix_set_color_all(0, 0, 0);
+            break;
+        case RGB_MODE_NONE:
+            rgb_matrix_set_flags(LED_FLAG_NONE);
+            rgb_matrix_set_color_all(0, 0, 0);
+            break;
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case RGB_TOG:
+            if (record->event.pressed) {
+                switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER | LED_FLAG_INDICATOR);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                        user_config.rgb_mode = RGB_MODE_KEYLIGHT;
```

Comment:
I'm not sure that this is needed, as the flags should be saved to eeprom, in general.

Reply:
Thanks, I've just pushed new commit to resolved this